### PR TITLE
fix: Stack Overflow plugin returns correct questions with relevant tags

### LINKED ISSUE #22097: 🐛 Bug Report: Stack Overflow plugin doesn't show questions with tags provided
### 📜 Description

The `<HomePageStackOverflowQuestions />` component doesn't show questions related to the tags provided in the `requestParams` object. 
This is because the `listQuestions` method in `StackOverflowApi`, specifically the `qs.stringify` part is prefixing the query parameters with indices, which makes the Stack Exchange API ignore the provided tag parameters.

image

This can be solved by adding the `arrayFormat: 'repeat'` option to `qs.stringify`, which will remove the prefix indexing and the API will successfully return questions related to the tags.

### 👍 Expected behavior

The API should return questions related to the tags provided to the component.

### 👎 Actual Behavior with Screenshots

image
image
image

Tags provided to the component are `backstage` and `techdocs` but none of the tags show up in the returned questions.

### 👟 Reproduction steps

Install `"@backstage/plugin-stack-overflow"` and add the `<HomePageStackOverflowQuestions />` component to the HomePage. 

Provide the component with a `requestParams` prop, e.g: 
`requestParams={{ tagged: ['backstage', 'techdocs'], pagesize: 5,
            site: 'stackoverflow'
          }}
`

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by OSfrog**
## Hey, I just made a Pull Request!

Adds the changes necessary to the "@backstage/plugin-stack-overflow" plugin to return questions related to the tags provided. See #22097 

Closes #22097 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/stack-overflow/src/api/StackOverflowClient.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export class StackOverflowClient implements StackOverflowApi {
   }): Promise<StackOverflowQuestion[]> {
     const params = qs.stringify(options.requestParams, {
       addQueryPrefix: true,
+      arrayFormat: 'repeat',
```
Comment (emmaindal) [inline plugins/stack-overflow/src/api/StackOverflowClient.ts:49]: would you be able to create a test for this use case in `plugins/stack-overflow/src/api/StackOverflowClient.test.ts` with the expected result?

**Review by emmaindal — COMMENTED** [review]
thank you for looking into this, just a small comment from my side!

File: plugins/stack-overflow/src/api/StackOverflowClient.ts (hunk 1 above)
Comment (OSfrog) [reply plugins/stack-overflow/src/api/StackOverflowClient.ts:49]: Reworked the test!

**Review by emmaindal — APPROVED** [review]
awesome, thanks a lot!

**Comment by emmaindal** [comment]
@backstage/maintainers the e2e test seems to fail on unrelated things, can you take a look and get this merged if e2e test looks ok? Thank you!

**Comment by OSfrog** [comment]
Please reopen this PR and let me know if anything else needs to be done for it to be merged!

**Review by freben — APPROVED** [review]
Oh wow sorry, completely missed the ping here.

**Pull request merged**